### DISCUSSION TITLE
Intention of the refresh token auth flow

### DISCUSSION THREAD

**Discussion by fue-medatixx** [Digitaler-Impfnachweis/certification-apis#40] [Q&A] [answered]
We should use the refresh token flow after the access token is expired. 
With "refresh" we get back another full token set which is a new access token and a new refresh token. 
So the previous steps repeat endless which means that a whole new access token (with new challenge and new smcb-signature) is only requested the first time after pvs system reboot. 

Is this the intention?

**Comment by mge-medisoftware** [comment]
No, the refresh only works for actual 1800 seconds after initiating authentication-flow. Because of TI RU refresh_expires_in =1800 
So every 30 minutes start from the beginning...

**Reply by fue-medatixx** [reply]
But with the refresh request I get back a different refresh token than I sent for the refresh process and again refresh_expires_in 1800. I interpret this as a new refresh token that is valid again for 1800 seconds.

**Reply by mge-medisoftware** [reply]
The different access token is a good argument, but the reason for that could be  security... (single use of refresh-token). Hmmm, so users should produce certificates at least every 30 minutes?  Or some thread doing the refresh. :-)  
Well, we can try if refresh is working after 30 minutes.

**Reply by mge-medisoftware** [reply]
I tried this. You were right.
14:00 authentication flow started
14:22 refresh worked
14:42 refresh worked (more then 30 minutes after start authentication flow)
15:47 refresh failed (http-state 400, more then 30 minutes after last refresh)

**Reply by fue-medatixx** [reply]
Thank you for testing it.
I hope for an answer from @oliver-steinbrecher  whether it doesn't matter in which way the refresh flow is implemented. As you have interpreted it or as it is currently possible, as an "endless" refresh

**Comment by oliver-steinbrecher** [comment, marked as answer]
The intention is to perform the smcb authentication once a day.
